SU2_CFD/src/solver_direct_mean.cpp

```diff
@@ -173,16 +173,171 @@ CEulerSolver::CEulerSolver(CGeometry *geometry, CConfig *config, unsigned short
 	bool time_stepping = config->GetUnsteady_Simulation() == TIME_STEPPING;
 	bool roe_turkel = (config->GetKind_Upwind_Flow() == TURKEL);
   bool adjoint = (config->GetContinuous_Adjoint()) || (config->GetDiscrete_Adjoint());
+  su2double AoA_, AoS_, iH_, dCL_dAlpha_, dCM_diH_, BCThrust_;
   string filename = config->GetSolution_FlowFileName();
+  string::size_type position;
+  unsigned long ExtIter_;
   bool rans = ((config->GetKind_Solver() == RANS )|| (config->GetKind_Solver() == DISC_ADJ_RANS));
-
   unsigned short direct_diff = config->GetDirectDiff();
   unsigned short nMarkerTurboPerf = config->Get_nMarkerTurboPerf();
   int rank = MASTER_NODE;
 #ifdef HAVE_MPI
   MPI_Comm_rank(MPI_COMM_WORLD, &rank);
 #endif
   
+  /*--- Check for a restart file to evaluate if there is a change in the angle of attack
+   before computing all the non-dimesional quantities. ---*/
+  
+  if (!(!restart || (iMesh != MESH_0) || nZone > 1)) {
+    
+    /*--- Modify file name for an unsteady restart (check this part...) ---*/
+    
+    if (dual_time) {
+      
+      if (adjoint) { Unst_RestartIter = SU2_TYPE::Int(config->GetUnst_AdjointIter()) - 1; }
+      else if (config->GetUnsteady_Simulation() == DT_STEPPING_1ST)
+        Unst_RestartIter = SU2_TYPE::Int(config->GetUnst_RestartIter())-1;
+      else
+        Unst_RestartIter = SU2_TYPE::Int(config->GetUnst_RestartIter())-2;
+      
+      filename = config->GetUnsteady_FileName(filename, Unst_RestartIter);
+    }
+    
+    /*--- Open the restart file, throw an error if this fails. ---*/
+    
+    restart_file.open(filename.data(), ios::in);
+    if (restart_file.fail()) {
+      if (rank == MASTER_NODE)
+        cout << "There is no flow restart file!! " << filename.data() << "."<< endl;
+      exit(EXIT_FAILURE);
+    }
+    
+    unsigned long iPoint_Global = 0;
+    string text_line;
+    
+    /*--- The first line is the header (General description) ---*/
+    
+    getline (restart_file, text_line);
+    
+    /*--- Space for the solution ---*/
+    
+    for (iPoint_Global = 0; iPoint_Global < geometry->GetGlobal_nPointDomain(); iPoint_Global++ ) {
+      
+      getline (restart_file, text_line);
+      
+    }
+    
+    /*--- Space for extra info (if any) ---*/
+    
+    while (getline (restart_file, text_line)) {
```

Comment:
If the portion of the constructor that reads the restarts is growing, it might be time to finally write a LoadRestart() method for all solver classes so that we can keep the constructors from getting too messy. I've always wanted to do this anyway (we have one for the mean flow), and it would keep duplicate code to a minimum.

Not necessarily for this PR, of course, but I would like to see this.